test/experiment.test.ts

```diff
@@ -0,0 +1,84 @@
+import { createClient } from '../src/client';
+import { LocalExperiment } from '../src/experiment';
+
+describe('LocalExperiment', () => {
+	test('define bad fractions', () => {
+		expect.assertions(2);
+		try {
+			const exp = new LocalExperiment<'var1' | 'var2'>('test');
+			exp.define('var1', 50).define('var2', 51);
+		} catch (e) {
+			expect(e.message).toContain('var1');
+			expect(e.message).toContain('var2');
+		}
+	});
+
+	test('define duplicated variations', () => {
+		expect.assertions(1);
+		try {
+			const exp = new LocalExperiment<'var1' | 'var2'>('test');
+			exp.define('var1', 50).define('var1', 30);
+		} catch (e) {
+			expect(e.message).toContain('var1');
+		}
+	});
+
+	test('missing variations', () => {
+		expect.assertions(1);
+		const exp = new LocalExperiment<'var1' | 'var2'>('test').define('var1', 99);
+		try {
+			exp.engage('test-user');
+		} catch (e) {
+			expect(e.message).toContain('fully');
+		}
+	});
+
+	describe('two variations experiment', () => {
+		const twoVariants = new LocalExperiment<'var1' | 'var2'>('test');
+		twoVariants.define('var1', 50).defineDefault('var2');
```

Comment:
@sradevski This is a proposal on how we can define the GSG modal experiment. It should be sufficient to run this experiment backed it with localStorage. And we can later hook Optimizely or our own backend implementation.

```
const gsgExp = new LocalExperiment<'ModalPresented'|'NoModal'>('GettingStartedGuide', analyticsClient);
const variation = gsgExp.engage(user.username);
if (variation === 'ModalPresented') {
  showModal();
}
```

Reply:
Yeah I think this is reasonable and simple enough to get us started quickly.